tensorflow/python/ops/array_ops.py

```diff
@@ -1757,6 +1757,176 @@ def _DepthToSpaceShape(op):
       [input_shape[0], height, width, new_depth])]
 
 
+def one_hot(indices, depth, on_value=None, off_value=None,
+            axis=None, dtype=None, name=None):
```

Comment:
Oh, I see, the crazy case analysis structure is so that the defaults can be 1.0 and 0.0 but setting just one of them to an integer makes the whole thing an integer.  Gah.  Let me think to see if there's a better way to do this.